public/js/graph/react_graph.js

```diff
@@ -554,6 +554,11 @@ var Graph = React.createClass({
         }
     },
 
+    drawNode: function(nodeJSON) {
+        var newNodesJSON = $.extend([], this.state.nodesJSON).push(nodeJSON);
+        this.setState({nodesJSON: newNodesJSON});
+    },
+
```

Comment:
Yeah, you'll need to extract data from the mouse event object. Take a look at how nodes are currently added.